snap/local/stage/bin/nat44_install

```diff
@@ -0,0 +1,44 @@
+
+#!/bin/bash -ex
+#
+#  Copyright (c) 2017, The OpenThread Authors.
+#  Copyright (c) 2023, Canonical Ltd.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+
+# WLAN_IFNAMES="${INFRA_IF_NAME:-eth0}"
+# THREAD_IF="${THREAD_IF:-wpan0}"
+export WLAN_IFNAMES=$(snapctl get infra-if)
+export THREAD_IF=$(snapctl get thread-if)
+
+# Just a random fwmark bits.
+iptables -t mangle -A PREROUTING -i $THREAD_IF -j MARK --set-mark 0x1001 -m comment --comment "OTBR NAT44"
+iptables -t nat -A POSTROUTING -m mark --mark 0x1001 -j MASQUERADE -m comment --comment "OTBR NAT44"
```

Comment:
Could you please explain why these commands have been added? None of the same commands could be found in the original OTBR upstream code regarding firewall setup.

Reply:
They are here: https://github.com/openthread/ot-br-posix/blob/1a84a172c96a531f03736a1f8f10ca89853e4545/script/_nat64#L160-L162

Reply:
Thanks, I now understand why I couldn't find them. It's because these items were recently added by this [PR](https://github.com/openthread/ot-br-posix/pull/1884) and my local git used the tag "thread-reference-20230119," which didn't include them.

Reply:
Interesting catch. I was looking at main while the source in snap is behind. This particular difference appears harmless. I hope there are not other unexpected differences.